### DISCUSSION TITLE
How to point the ref in validation object to the first input of a fieldArray?

### DISCUSSION THREAD

**Discussion by zorzysty** [react-hook-form/react-hook-form#5196] [Q&A] [answered]
Here's my yup schema:
```js
const schema = yup.object().shape({
  dropType: yup.string().required(),
  coCreators: yup
    .mixed()
    .test(
      'co-creators sum',
      'Sum of the co-creator shares must equal 100%',
      (coCreators) => {
        return coCreators.reduce((accu, curr) => accu + curr.share, 0) === 100
      }
    ),
})
```
`coCreators` is a fieldArray:
```js
  const { fields } = useFieldArray({ control, name: 'coCreators' })
```

Where each field looks like that:
```js
{
  id: number,
  share: number,
}
```

So when the sum of shares is not equal to 100, I'm getting this error object:
```js
​{
  coCreators: { 
    message: "Sum of the co-creator shares must equal 100%",
    type: "co-creators sum",
    ref: undefined 
  }
}
```
The problem here is that `ref` is `undefined`, while I would like it to point to the first input of this fieldArray. Is this possible?

**Comment by bluebill1049** [comment]
you will need to manage the ref yourself, there is no particular input which result in this error right?

**Reply by zorzysty** [reply]
Correct, no particular single input is responsible for this error, but rather all the inputs within the fieldArray together. That's why I would like the ref to point to the first input of this fieldArray.

**Comment by leapful** [comment, marked as answer]
@zorzysty @bluebill1049 

There is param "options" of callback "resolver" which already includes the fields reference therefore it can be used to implement a custom resolver and apply any handling custom "errors" object.

For example:
```

useForm({
   resolver:  async (values, context, options) => {
       const { fields } = options;
       try {
           await schema.validate(values, { abortEarly: false });
       } catch (e) {
          // e.inner is { path: string, message: string, type: string  }[]
          // just prepare errors object for "coCreators" with fields[path][0].share.ref like
         // { coCreators: {  message: 'message from e.inner item', ref: fields[path][0].share.ref, type: 'type from e.inner item' } }
       }
   }

})

```

Working sample: https://codesandbox.io/s/react-hook-form-validationschema-v7-forked-jlt5k?file=/src/index.js

**Reply by zorzysty** [reply]
This works. Thank you @leapful!

**Reply by bluebill1049** [reply]
thank you @leapful!
